### DISCUSSION TITLE
BOLT Optimizations in the future looks promising...

### DISCUSSION THREAD

**Discussion by Grimish-ng** [Frogging-Family/linux-tkg#965] [Ideas]
Claiming 5% performance boost with BOLT optimizations. 
Maybe work this into the build if that infact is remotely possible? Seems promising!
Phoronix link: https://www.phoronix.com/news/Linux-BOLT-5p-Performance

Of course this all depends on the workloads and I'm not sure about what workloads may have these types of gains but thats exactly the purpose of the discussion.

LLVM guide for this: https://github.com/llvm/llvm-project/blob/main/bolt/docs/OptimizingLinux.md

Just a suggestion if it does pan out.

**Comment by sebastian-de** [comment]
Support for Propeller got merged for Linux 6.13 as well: https://docs.kernel.org/dev-tools/propeller.html

But PGO is complicated - I'm sceptical if it's possible to generate good profiles that provide benefits for desktop/gaming use at all.

If you're interested, here is a (WIP) PGO article with a lot of details and links: https://github.com/zamazan4ik/awesome-pgo/blob/main/article/article.md

**Reply by ptr1337** [reply]
https://github.com/CachyOS/linux-cachyos/pull/342

Here you can see how we handle it in CachyOS. The user still needs to manually modify the PKGBUILD and set the env variable, and also needs to do its own profiling.

Propellor is pending, see:
https://github.com/CachyOS/linux-cachyos/issues/343
